### PR TITLE
Update docs: Added missing user methods and included API signatures. Added postAsBody.

### PR DESCRIPTION
**Pull request by andykirk**
Hi,
First step/attempt at improving the docs.
Let me know what you think. Happy to discuss and amend here.

Cheers

### CODE REVIEW

**Comment by andykirk** [comment]
Added postAsBody method and updated addCustomFieldOption. Things now work as expected.

**Comment by andykirk** [comment]
More additions to the docs.

File: src/Http.php [hunk 1]
```diff
@@ -213,6 +213,24 @@ public function post($path, $parameters)
         return $this->sendRequest($request);
     }
 
+    /**
+     * Retrieves http response for a request with the put method,
```
Comment (stevenmaguire) [inline src/Http.php]: "Retrieves http response for a request with the **_post_** method," - documentation needs to reflect http method

File: API-GUIDE.md [hunk 2]
```diff
@@ -2,99 +2,246 @@
...
+
+#### Get Current User
+
+```php
+$result = $client->getCurrentUser();
+```
+
+API Signature:
+```php
+'getCurrentUser' => ['get', 'members/me']
```
Comment (stevenmaguire) [inline API-GUIDE.md]: I love the idea of including the http method and actual trello api signature. I think including this odd snippet from the ApiMethodsTrait maybe confusing as PHP code. It sort of looks like you need to pass that array as arguments or something. Can we update this format to be plaintext like....

`GET /members/me`
`GET /actions/{actionId}`

... something a bit less technical and still conveying the important information.

**Review by stevenmaguire — CHANGES_REQUESTED** [review]

**Comment by andykirk** [comment]
Hi,
Apologies for the put/post typo.
I'll fix this PR ASAP.

File: API-GUIDE.md (hunk 2 above)
Comment (andykirk) [reply API-GUIDE.md]: I see. I was quoting directly from the PHP code, hence the method used.
I'll change that.

**Comment by andykirk** [comment]
Phew! That was a bigger job than I first envisaged.
I'm on the fence about the inclusion of the 'API Signature:' wording.
Is it still clear what this is without it?
I can easily change or remove that wording though.

**Pull request closed without merging**